modelon/impact/client/options.py

```diff
@@ -51,5 +62,82 @@ def with_values(self, **modified):
             sol_opts = custom_function.get_solver_options().with_values(rtol=1e-7)
             sim_opts = custom_function.get_simulation_options().with_values(ncp=500)
         """
-        values = _set_options(self._values, **modified,)
-        return ExecutionOptions(values, self._name, self._custom_func_sal)
+        values = _set_options(self._values, **modified)
+        return self.data(values)
```

Comment:
I think this could be simplified a lot by instead of having an abstract method and all you would use:
`return self.__class__(values, self._custom_function_name)` which would return the correct class instance.